llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -463,12 +466,16 @@ int PostPipeliner::mostUrgent(PostPipelinerStrategy &Strategy) {
   return Best;
 }
 
-void PostPipeliner::resetSchedule() {
+void PostPipeliner::resetSchedule(bool ResetCritical) {
   Scoreboard.clear();
   for (int K = 0; K < NTotalInstrs; K++) {
     auto &N = Info[K];
-    N.LastEarliestPusher = -1;
-    N.LastLatestPusher = -1;
+    if (ResetCritical) {
+      N.NumPushedEarliest = 0;
+      N.NumPushedLatest = 0;
+      N.LastEarliestPusher = -1;
+      N.LastLatestPusher = -1;
+    }
```

Comment:
These parameters represents the path that pushed Earliest of the failing node beyond its modulo cycle in the previous iteration. If we don't reset it, we can use it to provide look-ahead in a rerun.